tests/providers/amazon/aws/operators/test_base_aws.py

```diff
@@ -171,3 +177,40 @@ class SoWrongOperator(AwsBaseOperator):
         error_match = r"Class attribute 'SoWrongOperator.aws_hook_class' is not a subclass of AwsGenericHook"
         with pytest.raises(AttributeError, match=error_match):
             SoWrongOperator(task_id="fake-task-id")
+
+    @pytest.mark.parametrize(
+        "region, region_name, expected_region_name",
+        [
+            pytest.param("ca-west-1", None, "ca-west-1", id="region-only"),
+            pytest.param("us-west-1", "us-west-1", "us-west-1", id="non-ambiguous-params"),
```

Comment:
What would happen for a test case where both are provided but different? Would `region_name` win-out in if partial is being used and `region` otherwise?

You might want to add a check in src to ensure they're the same if so.

Reply:
will fail

https://github.com/apache/airflow/blob/dec2662190dd4480d0c631da733e19d2ec9a479d/airflow/providers/amazon/aws/utils/mixins.py#L95-L98

Reply:
This work exactly as it work before, in this PR is is a just a fix for a MappedOperator

Reply:
Glad to hear the case is covered already. I think it's still worth adding a test case for the mapped case IMHO, but up to you.

Reply:
I work for fake operators based on AwsBaseOperator / AwsBaseSensor which do simple actions nothing on execute.
For EcsRunTaskOperator it also works, however it required to do additional thing just for not break other tests.
I still tried to find a better way to do that kind of test, the last resort would be a mock as we do for other test.